ResearchKit/Common/ORKNavigableOrderedTask.m

```diff
@@ -191,4 +195,133 @@ - (NSUInteger)hash {
+            step.numberOfPegs = numberOfPegs;
+            step.threshold = threshold;
+            step.rotated = rotated;
+            step.shouldTintImages = YES;
+            step.stepDuration = timeLimit == 0 ? CGFLOAT_MAX : timeLimit;
+            
+            ORKStepArrayAddStep(steps, step);
+        }
+        
+        {
+            ORKHolePegTestRemoveStep *step = [[ORKHolePegTestRemoveStep alloc] initWithIdentifier:ORKHolePegTestDominantRemoveStepIdentifier];
+            step.title = [NSString stringWithFormat:ORKLocalizedString(@"HOLE_PEG_TEST_REMOVE_INSTRUCTION_%@", nil), dHand];
+            step.text = ORKLocalizedString(@"HOLE_PEG_TEST_TEXT", nil);
+            step.spokenInstruction = step.title;
+            step.orientation = dominantHand == ORKSideLeft ? ORKSideRight : ORKSideLeft;
+            step.dominantHandTested = YES;
+            step.numberOfPegs = numberOfPegs;
+            step.threshold = threshold;
+            step.shouldTintImages = YES;
+            step.stepDuration = timeLimit == 0 ? CGFLOAT_MAX : timeLimit;
+            
+            ORKStepArrayAddStep(steps, step);
+        }
+        
+        {
+            ORKHolePegTestPlaceStep *step = [[ORKHolePegTestPlaceStep alloc] initWithIdentifier:ORKHolePegTestNonDominantPlaceStepIdentifier];
+            step.title = [NSString stringWithFormat:ORKLocalizedString(@"HOLE_PEG_TEST_PLACE_INSTRUCTION_%@", nil), ndHand];
+            step.text = ORKLocalizedString(@"HOLE_PEG_TEST_TEXT", nil);
+            step.spokenInstruction = step.title;
+            step.orientation = dominantHand == ORKSideLeft ? ORKSideRight : ORKSideLeft;
+            step.dominantHandTested = NO;
+            step.numberOfPegs = numberOfPegs;
+            step.threshold = threshold;
+            step.rotated = rotated;
+            step.shouldTintImages = YES;
+            step.stepDuration = timeLimit == 0 ? CGFLOAT_MAX : timeLimit;
+            
+            ORKStepArrayAddStep(steps, step);
+        }
+        
+        {
+            ORKHolePegTestRemoveStep *step = [[ORKHolePegTestRemoveStep alloc] initWithIdentifier:ORKHolePegTestNonDominantRemoveStepIdentifier];
+            step.title = [NSString stringWithFormat:ORKLocalizedString(@"HOLE_PEG_TEST_REMOVE_INSTRUCTION_%@", nil), ndHand];
+            step.text = ORKLocalizedString(@"HOLE_PEG_TEST_TEXT", nil);
+            step.spokenInstruction = step.title;
+            step.orientation = dominantHand;
+            step.dominantHandTested = NO;
+            step.numberOfPegs = numberOfPegs;
+            step.threshold = threshold;
+            step.shouldTintImages = YES;
+            step.stepDuration = timeLimit == 0 ? CGFLOAT_MAX : timeLimit;
+            
+            ORKStepArrayAddStep(steps, step);
+        }
+    }
+    
+    if (! (options & ORKPredefinedTaskOptionExcludeConclusion)) {
+        ORKInstructionStep *step = [self makeCompletionStep];
+        
+        ORKStepArrayAddStep(steps, step);
+    }
+    
```

Comment:
I'd add a comment here explaining that the task is actually dynamic and that the *direct navigation rules* are used for skipping the peg removal steps if the user doesn't succeed in placing all the pegs in the allotted time (they rules removed from `ORKHolePegTestPlaceStepViewController` if she succeeds).